Libplanet/Blockchain/BlockChain.cs

```diff
@@ -1171,29 +1171,24 @@ internal void Swap(BlockChain<T> other, bool render)
             if (other?.Tip is null)
             {
                 throw new ArgumentException(
-                        $"The chain to be swapped is invalid. Id: {other?.Id}, Tip: {other?.Tip}",
-                        nameof(other));
+                    $"The chain to be swapped is invalid. Id: {other?.Id}, Tip: {other?.Tip}",
+                    nameof(other));
             }
 
             _logger.Debug(
                 "Swapping block chain. (from: {fromChainId}) (to: {toChainId})", Id, other.Id);
 
             // Finds the branch point.
             Block<T> topmostCommon = null;
-            if (render && !(Tip is null))
+            if (!(Tip is null))
             {
                 long shorterHeight =
                     Math.Min(Count, other.Count) - 1;
-                for (
-                    Block<T> t = this[shorterHeight], o = other[shorterHeight];
-                    t.PreviousHash is HashDigest<SHA256> tp &&
-                        o.PreviousHash is HashDigest<SHA256> op;
-                    t = this[tp], o = other[op]
-                )
+                for (long index = shorterHeight; index >= 0; --index)
                 {
-                    if (t.Equals(o))
+                    if (this[index].Equals(other[index]))
```

Comment:
How about using `Block<T>.PreviousHash` instead of random access?

Reply:
It was to use `index`, not `Block<T>.PreviousHash` because I didn't have any idea for how to traverse, including the genesis block. Do you have some ideas for it?